### PR TITLE
Refactor/async model training

### PR DESCRIPTION
**Pull request by BrettBoval**
Refactor to make the ensemble generation into a more standard unix pipeline. Still WIP.

### Please Check
- [ ] Are the [JSON schemas](https://github.com/covid-projections/covid-data-model/tree/master/api/schemas) updated?
- [ ] Are the [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.V1.md) updated?
- [ ] Are tests passing?

### CODE REVIEW

File: pyseir/ensembles/ensemble_runner_v2.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        if key not in ("t_list", "county_metadata")
+    }
+    for model in model_ensemble:
+        for key in compartments:
+            compartments[key].append(model.results[key])
+
+    return {key: np.vstack(value_stack) for key, value_stack in compartments.items()}
+
+
+def _get_surge_window(model_ensemble, compartment):
```
Comment (ghop02) [inline pyseir/ensembles/ensemble_runner_v2.py:80]: We don't need surge windows anymore do we?

File: pyseir/ensembles/ensemble_runner_v2.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            m.results[compartment][::-1] > getattr(m, compartment_to_capacity_attr_map[compartment])
+        )
+        surge_end.append(
+            m.t_list[::-1][surge_end_idx[0][0]] if len(surge_end_idx) > 0 else float("NaN")
+        )
+
+    return surge_start, surge_end
+
+
+def _detect_peak_time_and_value(value_stack, t_list):
```
Comment (ghop02) [inline pyseir/ensembles/ensemble_runner_v2.py:122]: Don't need this either right?

File: pyseir/ensembles/ensemble_runner_v2.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        # Merge this dictionary into the suppression level one.
+        outputs[compartment].update(compartment_output)
+
+    return outputs
+
+
+if __name__ == "__main__":
+    print("Running Main")
+    example_inputs = {
+        "fips": "16",
```
Comment (ghop02) [inline pyseir/ensembles/ensemble_runner_v2.py:208]: would prefer to separate out fips from the rest of the inputs.  

And could you make the other inputs a dataclass? 😍

File: pyseir/inference/model_fitter.py [hunk 4]
```diff
@@ -406,7 +420,12 @@ def run_model(self, R0, eps, t_break, eps2, t_delta_phases, log10_I_initial):
         """
 
         suppression_policy = suppression_policies.get_epsilon_interpolator(
-            eps, t_break, eps2, t_delta_phases
+            eps=eps,
+            t_break=t_break,
```
Comment (ghop02) [inline pyseir/inference/model_fitter.py:424]: Prefer not to call these with named arguments unless they're keyword arguments

File: pyseir/ensembles/ensemble_runner_v2.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        peak_data[f"peak_value_ci{percentile}"] = np.percentile(
+            values_at_peak_index, percentile
+        ).tolist()
+        peak_data["peak_time_ci{percentile}"] = np.percentile(peak_times, percentile).tolist()
+
+    peak_data["peak_value_mean"] = np.mean(values_at_peak_index).tolist()
+    return peak_data
+
+
+def _generate_output_for_suppression_policy(model_ensemble):
```
Comment (ghop02) [inline pyseir/ensembles/ensemble_runner_v2.py:158]: Can you type the inputs and outputs here?

**Comment by BrettBoval** [comment]
This code change was superseded by changes made in pipeline refactoring.

**Pull request closed without merging**